README.rst
=============
aldryn-disqus
=============

Integrate `Disqus <http://disqus.com>`_ into your
`django CMS <http://django-cms.org>`_ and `Aldryn projects <http://aldryn.com>`_,
and allow users to comment on and discuss content on your site.

Disqus is one of the most popular commenting systems available. It’s especially
suited to news and weblog content, but can be applied anywhere that you’d like
to provide discussion functionality.

With the Aldryn Disqus Addon you can integrate it into your projects and start
building an online community in just a few simple steps.

`See it in action on the django CMS weblog <http://www.django-cms.org/en/blog/>`_.

Installation
============

Aldryn Installation
-------------------

Choose a site you want to install the add-on to from the dashboard. Then go to
``Apps -> Install app`` and click ``Install`` next to ``Aldryn Disqus`` app.

Ensure you correctly set the ``DISQUS_SHORTNAME`` setting in the control panel
to the identifier you configured for your project at Disqus.

Redeploy the site.


Manual Installation
-------------------

Add 'aldryn_disqus' to your project's settings and run ``python manage.py migrate``.

Add ``DISQUS_SHORTNAME = 'projectname'`` to your settings. Where ``projectname``
is the identifier you configured for your project at Disqus.


Usage
-----

After configuring your Disqus account at https://disqus.com/, simply add the
``Aldryn Disqus`` plugin on the desired page(s) in your project.

CHANGELOG.rst
Changelog
=========

0.1.1 (2015-10-30)
------------------

* Includes south_migrations in manifest


0.1.0 (2015-10-15)
------------------

* Adds Django 1.7+ support


0.0.11 (2015-10-14)
-------------------

[ Previous releases ]


aldryn_disqus/south_migrations/0001_initial.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):

        # Adding model 'DisqusPlugin'
        db.create_table('cmsplugin_disqusplugin', (
            ('cmsplugin_ptr', self.gf('django.db.models.fields.related.OneToOneField')(to=orm['cms.CMSPlugin'], unique=True, primary_key=True)),
        ))
        db.send_create_signal('aldryn_disqus', ['DisqusPlugin'])


    def backwards(self, orm):

        # Deleting model 'DisqusPlugin'
        db.delete_table('cmsplugin_disqusplugin')


    models = {
        'cms.cmsplugin': {
            'Meta': {'object_name': 'CMSPlugin'},
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'language': ('django.db.models.fields.CharField', [], {'max_length': '15', 'db_index': 'True'}),
            'level': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'lft': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'parent': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.CMSPlugin']", 'null': 'True', 'blank': 'True'}),
            'placeholder': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.Placeholder']", 'null': 'True'}),
            'plugin_type': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'}),
            'position': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True', 'blank': 'True'}),
            'rght': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'tree_id': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'})
        },
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'})
        },
        'aldryn_disqus.disqusplugin': {
            'Meta': {'object_name': 'DisqusPlugin', 'db_table': "'cmsplugin_disqusplugin'", '_ormbases': ['cms.CMSPlugin']},
            'cmsplugin_ptr': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['cms.CMSPlugin']", 'unique': 'True', 'primary_key': 'True'}),
        }
    }

    complete_apps = ['aldryn_disqus']


aldryn_disqus/south_migrations/0002_cmsplugin_table_name_magic_removal.py
# -*- coding: utf-8 -*-
from south.utils import datetime_utils as datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


# copied from
# https://github.com/stefanfoulis/cmsplugin-filer/blob/a0a2119907c41f85303b069fe03d346d15bb0741/cmsplugin_filer_utils/migration.py
def rename_tables(db, table_mapping, reverse=False):
    """
    renames tables from source to destination name, if the source exists and the destination does
    not exist yet.
    """
    from django.db import connection
    if reverse:
        table_mapping = [(dst, src) for src, dst in table_mapping]
    table_names = connection.introspection.table_names()
    for source, destination in table_mapping:
        if source in table_names and destination in table_names:
            print(u"    WARNING: not renaming {0} to {1}, because both tables already exist.".format(source, destination))
        elif source in table_names and destination not in table_names:
            print(u"     - renaming {0} to {1}".format(source, destination))
            db.rename_table(source, destination)


def rename_tables_old_to_new(db, table_mapping):
    return rename_tables(db, table_mapping, reverse=False)


def rename_tables_new_to_old(db, table_mapping):
    return rename_tables(db, table_mapping, reverse=True)

# end copied from


class Migration(SchemaMigration):
    cms_plugin_table_mapping = (
        # (old_name, new_name),
        ('cmsplugin_disqusplugin', 'aldryn_disqus_disqusplugin'),
    )

    def forwards(self, orm):
        rename_tables_old_to_new(db, self.cms_plugin_table_mapping)

    def backwards(self, orm):
        rename_tables_new_to_old(db, self.cms_plugin_table_mapping)

    models = {
        u'aldryn_disqus.disqusplugin': {
            'Meta': {'object_name': 'DisqusPlugin', '_ormbases': ['cms.CMSPlugin']},
            u'cmsplugin_ptr': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['cms.CMSPlugin']", 'unique': 'True', 'primary_key': 'True'})
        },
        'cms.cmsplugin': {
            'Meta': {'object_name': 'CMSPlugin'},
            'changed_date': ('django.db.models.fields.DateTimeField', [], {'auto_now': 'True', 'blank': 'True'}),
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'depth': ('django.db.models.fields.PositiveIntegerField', [], {}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'language': ('django.db.models.fields.CharField', [], {'max_length': '15', 'db_index': 'True'}),
            'numchild': ('django.db.models.fields.PositiveIntegerField', [], {'default': '0'}),
            'parent': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.CMSPlugin']", 'null': 'True', 'blank': 'True'}),
            'path': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '255'}),
            'placeholder': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.Placeholder']", 'null': 'True'}),
            'plugin_type': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'}),
            'position': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True', 'blank': 'True'})
        },
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '255', 'db_index': 'True'})
        }
    }

    complete_apps = ['aldryn_disqus']
